packages/compiler/config/config-loader.ts

```diff
@@ -17,12 +18,19 @@ export const defaultConfig = deepFreeze({
 
 /**
  * Look for the project root by looking up until a `tspconfig.yaml` is found.
- * @param path Path to start looking
+ * @param path Path to the file or the folder to start looking
  */
 export async function findTypeSpecConfigPath(
   host: CompilerHost,
   path: string
 ): Promise<string | undefined> {
+  // if the path is a file, return immediately
+  if (fs.existsSync(path)) {
```

Comment:
I think you want to use the compiler host filesystem utils here

Reply:
Thanks updated.